### ISSUE TITLE
Testing

### ISSUE THREAD

**Issue by ahasselbring** [bhuman/CompiledNN#6]
At least the individual operation compilers need to be automatically tested for correctness. This has already been started for the `UpSampling2D` operation. A further step would be testing the merging of layers.

- [ ] Activation
- [ ] Arithmetic
- [ ] BatchNormalization
- [ ] Concatenate
- [ ] Conv2D
- [ ] Cropping2D
- [ ] DConv2D
- [ ] Dense
- [ ] GlobalPooling2D
- [ ] Im2Col2D
- [ ] Pooling2D
- [ ] Softmax
- [ ] UInt8Input
- [x] UpSampling2D
- [x] ZeroPadding2D

Summary of the original issue description in the  B-Human repository:

We do not know whether inference is always correct. Some bugs are known, but there are probably more that just do not occur in the models we use. We need systematic, unit-test-like tests which create all kinds of network architectures, apply them to random data and compare the result of CompiledNN to SimpleNN. The goal is 100% path coverage of all compiler classes, also considering that some layers can sometimes be fused. The tests should be a standalone executable in the CompiledNN repository. A nice addition would be the evaluation of execution times of different model configurations. E.g., once the implementations of `Im2Col` and `Conv1x1` are done, there need to be systematic comparisons between `Conv2D` and `Im2Col + Conv1x1`.
